### PR TITLE
[CFM-1188] Add subnavigation to help section

### PR DESCRIPTION
**Pull request by ybaras**
[CFM-1188](https://issuetracker.amplexor.com/jira/browse/CFM-1188)

### CODE REVIEW

**Comment by ordavidil** [comment]
@ybaras 
please add screenshots

**Comment by ybaras** [comment]
**Visitor:**

anon

**Site administrator:**

admin-ma

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.module [hunk 1]
```diff
@@ -148,6 +148,141 @@ function c4m_content_book_help_and_guidance_autocomplete($title = '') {
...
+ * new help pages.
+ */
+function c4m_content_book_node_presave($node) {
+  // We are not saving a book page.
+  if ($node->type != 'book' || empty($node->book['menu_name'])) {
+    return;
+  }
+
+  // We are not saving a help book page.
+  $main_help_page = c4m_content_book_get_main_help_and_guidance();
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.module]: validate the returned value.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.module [hunk 2]
```diff
@@ -148,6 +148,141 @@ function c4m_content_book_help_and_guidance_autocomplete($title = '') {
...
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function c4m_content_book_block_view($delta = '') {
+  $block = array();
+
+  switch ($delta) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.module]: use `if`

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.module [hunk 3]
```diff
@@ -148,6 +148,141 @@ function c4m_content_book_help_and_guidance_autocomplete($title = '') {
...
+    return;
+  }
+
+  // We are not saving a help book page.
+  $main_help_page = c4m_content_book_get_main_help_and_guidance();
+  if ($node->book['menu_name'] != $main_help_page->book['menu_name']) {
+    return;
+  }
+
+  $node->path['alias'] = 'help-guidance/' . c4m_og_human_to_machine_readable($node->title);
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.module]: Add ac omment

**Pull request merged**